### PR TITLE
Convert to React + Typescript

### PR DESCRIPTION
**Pull request by n7huang**
npm install to setup node modules
then npm run build to get a build directory
chrome://extensions/ -> Load Unpacked -> the build directory to import into chrome to test

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+
+it('renders without crashing', () => {
```
Comment (EssaSaeed) [inline src/App.test.js]: wonderful :)

Comment (n7huang) [reply src/App.test.js]: the start of something great

**Comment by n7huang** [comment]
`package-lock.json` should be checked in with source control.
see: https://github.com/npm/npm/blob/v5.0.0/doc/files/package-lock.json.md

**Comment by n7huang** [comment]
Will merge tomorrow morning before demo in case anyone has any objections :)

**Pull request merged**